### ISSUE TITLE
Why augment scale?

### ISSUE THREAD

**Issue by Jacobi93** [ultralytics/yolov5#2555]
## ❔Question
in yolov5/utils/autoanchor.py, i found
scale = np.random.uniform(0.9, 1.1, size=(shapes.shape[0], 1))  # augment scale
wh = torch.tensor(np.concatenate([l[:, 3:5] * s for s, l in zip(shapes * scale, dataset.labels)])).float()  # wh
in the check_anchors() function.

My understanding is that they add some noise to wh. Is it for the robustness of the model? Will the performance drop without this step?

## Additional context
Thank you so much for this amazing project.

**Comment by glenn-jocher** [comment]
@Jacobi93 this probably doesn't have much effect at it's current setting (10%). This adds 10% scale jitter to the image sizes used for autoanchor. The default hyp.scratch.yaml file uses 50% scale jitter during training, and of course no scale jitter is used during testing unless TTA is enabled.
https://github.com/ultralytics/yolov5/blob/e8fc97aa3891f05812d7dfff90ca66d3481bda2c/data/hyp.scratch.yaml#L27

**Comment by Jacobi93** [comment]
> @Jacobi93 this probably doesn't have much effect at it's current setting (10%). This adds 10% scale jitter to the image sizes used for autoanchor. The default hyp.scratch.yaml file uses 50% scale jitter during training, and of course no scale jitter is used during testing unless TTA is enabled.
> https://github.com/ultralytics/yolov5/blob/e8fc97aa3891f05812d7dfff90ca66d3481bda2c/data/hyp.scratch.yaml#L27

Thank you so much for your reply. I think the 10% is for the annotation of wh, not for image sizes. That is, it changes the size of the ground truth bounding boxes without changing the image sizes.
Yes, I noticed the 50% for the image scale. It should be different from the above step. I just wonder if the performance will drop without this step.
Thanks.

**Comment by glenn-jocher** [comment]
@Jacobi93 yes that's correct. I doubt the 10% has much effect, though it may also serve to separate unique points as sometimes the scipy kmeans function will fail to return the number of requested centroids if the data is underdetermined (i.e. if 1000 of the same point are passed it may have difficulty finding 9 kmeans).
